tests/ui/result_as_ref_deref.rs

```diff
@@ -0,0 +1,58 @@
+#![allow(unused, clippy::redundant_clone, clippy::useless_vec)]
```

Comment:
nit: could you add error markers (`//~^`) to lines which are supposed to shown an error? This will then be checked automatically for non-regressions, instead of having to visually inspect changes.

Reply:
Those would have to be added manually. The tests haven't all been updated to use them, but they are preferred for anything new. You can add `//~ lint_name` on the end of any line that should lint. e.g.

``` rust
let _ = res.clone().as_ref().map(Deref::deref).map(str::len) //~ result_as_ref_deref
```

Using `//~^ line_name` will let you add it to the following line, but that should only be used for longer lines.